expire-tiles.cpp

```diff
@@ -413,80 +413,71 @@ int expire_tiles::from_bbox(double min_lon, double min_lat, double max_lon, doub
 	return 0;
 }
 
-void expire_tiles::from_nodes_line(const struct osmNode * nodes, int count) {
-	int	i;
-	double	last_lat;
-	double	last_lon;
-
-	if (Options->expire_tiles_zoom < 0) return;
-	if (count < 1) return;
-	last_lat = nodes[0].lat;
-	last_lon = nodes[0].lon;
-	if (count < 2) {
-		from_bbox(last_lon, last_lat, last_lon, last_lat);
-		return;
-	}
-	for (i = 1; i < count; i ++) {
-		from_line(last_lon, last_lat, nodes[i].lon, nodes[i].lat);
-		last_lat = nodes[i].lat;
-		last_lon = nodes[i].lon;
-	}
+void expire_tiles::from_nodes_line(const nodelist_t &nodes)
+{
+    if (Options->expire_tiles_zoom < 0 || nodes.empty())
+        return;
+
+    if (nodes.size() == 1) {
+        from_bbox(nodes[0].lon, nodes[0].lat, nodes[0].lon, nodes[0].lat);
+    } else {
+        for (int i = 1; i < nodes.size(); ++i)
```

Comment:
In theory yes, but I found that version much more readable. With nodes being const, I trust that the compiler knows how to optimize the code correctly.